cpp/src/arrow/compute/kernels/codegen_internal.h

```diff
@@ -1134,8 +1340,43 @@ ArrayKernelExec GeneratePhysicalInteger(detail::GetTypeId get_id) {
   }
 }
 
-template <template <typename... Args> class Generator, typename... Args>
+template <template <typename... Args> class Generator, typename Type0, typename... Args>
 ArrayKernelExec GeneratePhysicalNumeric(detail::GetTypeId get_id) {
```

Comment:
Can you add comments to these two functions, to explain how they differ?

Reply:
Maybe something like:
```c++
// Generate a kernel given a templated functor for numeric types.
// The functor is expected to have a constant output type specified via its first template parameter.
GeneratePhysicalNumeric()

// Generate a kernel given a templated functor for numeric types.
// The functor is expected to have the same input and output numeric types.
GeneratePhysicalNumericToNumeric()
```

Reply:
Applied edponce suggestions.